Add methods in DepthProfile

MarekOchocki opened the pull request:
issue #4

MarekOchocki committed:
Add methods in DepthProfile

- range
- FWHM
- max_plat_ratio
- distal_falloff
- modulation
- normalize

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,36 @@
+        return self.width(0.5)
+
+    def max_plat_ratio(self):
+        return self.y.max()/self.y[self.x.min()]
```
indexing by x won't work this way.
Why not `self.y[0]` ?

MarekOchocki replied:
Fixed

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,36 @@
+    def range(self, level):
+        return self.width(level)
+
+    def FWHM(self):
```
rename to `fwhm`

MarekOchocki replied:
Done.

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,36 @@
-    """TODO"""
-    pass
+
+    def range(self, level):
```
range will collide with python `range` keyword. Think about better name.

MarekOchocki replied:
I removed this method, because it's just duplication of width() from Profile

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,36 @@
+        return self.x_at_y(0.1, True) - self.x_at_y(0.9, True)
+
+    def modulation(self, level):
+        if np.isnan(self.x_at_y(level)):
```
Add some comments explaining why is it implemented this way.

grzanka commented:
@MarekOchocki what about tests ?

MarekOchocki committed:
Add comment and rename some methods

- remove range, because function width() from Profile does the same
- explain modulation
- rename fwhm

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,37 @@
-    """TODO"""
-    pass
+
+    def fwhm(self):
```
Why are you re-implementing it ?
It is a method of a base class:

https://github.com/DataMedSci/beprof/blob/master/beprof/profile.py#L128

grzanka commented on ibodata/depth_profile.py:
```diff
@@ -1,6 +1,37 @@
 from beprof.profile import Profile
 
 
 class DepthProfile(Profile):
```
range method is missing

grzanka replied:
it would be stupid to name method calculating range as `range` as this would introduce conflict with python range keyword

what about naming it `ran` ?

MarekOchocki committed:
Add ran method and remove fwhm

fwhm is already implemented in Profile
